recipes/tcl_installer/8.6.10/conanfile.py

```diff
@@ -0,0 +1,202 @@
+            opts.append("nomsvcrt")
+        if "d" not in self.settings.compiler.runtime:
+            opts.append("unchecked")
+        vcvars_command = tools.vcvars_command(self.settings)
+        self.run(
+            '{vcvars} && nmake -nologo -f "{cfgdir}/makefile.vc" INSTALLDIR="{pkgdir}" OPTS={opts} {targets}'.format(
+                vcvars=vcvars_command,
+                cfgdir=self._get_configure_dir("win"),
+                pkgdir=self.package_folder,
+                opts=",".join(opts),
+                targets=" ".join(targets),
+            ), cwd=self._get_configure_dir("win"),
+        )
+
+    def _build_autotools(self):
+        conf_args = [
+            "--enable-threads",
+            "--enable-shared" if self.options.shared else "--disable-shared",
+            "--enable-symbols" if self.settings.build_type == "Debug" else "--disable-symbols",
+            "--enable-64bit" if self.settings.arch_build == "x86_64" else "--disable-64bit",
+            "--prefix={}".format(self.package_folder),
+        ]
+        autoTools = self._get_auto_tools()
+        autoTools.configure(configure_dir=self._get_configure_dir(), args=conf_args, vars={"PKG_CFG_ARGS": " ".join(conf_args)})
+
+        # https://core.tcl.tk/tcl/tktview/840660e5a1
+        for root, _, files in os.walk(self.build_folder):
+            if "Makefile" in files:
+                tools.replace_in_file(os.path.join(root, "Makefile"), "-Dstrtod=fixstrtod", "", strict=False)
+
+        with tools.chdir(self.build_folder):
+            autoTools.make()
+
+    def build(self):
+        self._fix_sources()
+        if self.settings.compiler == "Visual Studio":
+            # do not treat warnings as errors 
+            tools.replace_in_file(os.path.join(self._source_subfolder, "win", "rules.vc"), "cwarn = $(cwarn) -WX", "")
+            self._build_nmake(["release"])
+        else:
+            self._build_autotools()
+
+    def package(self):
+        if self.settings.compiler == "Visual Studio":
+            self._build_nmake(["install-binaries", "install-libraries"])
+            # there is no way to only install headers + libraries
+            # hence why we build/install everything and then remove
+        else:
+            with tools.chdir(self.build_folder):
+                autoTools = self._get_auto_tools()
+                autoTools.make(target="install-binaries")
+                autoTools.make(target="install-libraries")
+                autoTools.make(target="install-msgs")
+                autoTools.make(target="install-tzdata")
+                autoTools.make(target="install-headers")
+                autoTools.make(target="install-private-headers")
+            pkgconfig_dir = os.path.join(self.package_folder, "lib", "pkgconfig")
+            tools.rmdir(pkgconfig_dir)
+        self.copy(pattern="license.terms", dst="licenses", src=self._source_subfolder)
+
+        tclConfigShPath = os.path.join(self.package_folder, "lib", "tclConfig.sh")
```

Comment:
This can also be moved to `_fix_sources` if there exists a `tclConfig.sh.in` file. But again, I'm not sure!!!

Reply:
Also, make sure that these environment variables do not clash. `tcl` vs `libtcl`

Reply:
(I was wrong about moving to `_fix_sources`. This is configure by the autotools script.)